Explorer/Assets/DCL/LOD/Components/SceneLODInfo.cs

```diff
@@ -39,26 +38,25 @@ public void Dispose(World world)
 
         public void SetCurrentLOD(LODAsset newLod)
         {
-            if (newLod.LoadingFailed)
-                ProfilingCounters.Failling_LOD_Amount.Value++;
-            else
+            CurrentLOD = newLod;
+            UpdateLastSuccessfullLOD();
+        }
+
+        public void UpdateLastSuccessfullLOD()
+        {
+            if (CurrentLOD.State == LODAsset.LOD_STATE.SUCCESS)
             {
-                if (!newLod.LodKey.Equals(LastSuccessfulLOD))
-                {
-                    LastSuccessfulLOD?.Release();
-                    LastSuccessfulLOD = newLod;
-                }
+                LastSuccessfulLOD?.Release();
+                LastSuccessfulLOD = CurrentLOD;
             }
-
-            CurrentLOD = newLod;
         }
 
-        public LODAsset? GetCurrentLOD()
+        public LODAsset GetCurrentLOD()
```

Comment:
`LODAsset` can be a `struct` (especially considering you keep it in this `struct`).
You would need to return it by `ref` but it's not allowed in C# to do it from the structure.
Thus, I would not bother about methods but make `CurrentLOD` and `LastSuccessfulLOD` `public`: anyway you need both read and write access.
Then you will use like this `ref var currentLOD = ref sceneLODInfo.CurrentLOD;`